<launch>
    <arg name="robot" default="mybot"/>
    <arg name="aruco_reacher" default="simple"/>
    <!-- 
        simple - with topic interface
        action - with action interface
        search - rotate and reach marker
        <any other> - disable aruco reaching
    -->

    <!-- Rotation action server -->
    <node pkg="mybot_2dnav" type="rotation_action_server.py" name="rotation_action_server" output="screen">
        <param name="default_speed" value="0.6"/>
        
    </node>
    
    <!-- THE MOVE BASE ITSELF -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        
        <rosparam file="$(find mybot_2dnav)/config/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
        
        <rosparam file="$(find mybot_2dnav)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        
        <rosparam file="$(find mybot_2dnav)/config/local_costmap_params.yaml" command="load" />
        
        <rosparam file="$(find mybot_2dnav)/config/global_costmap_params.yaml" command="load" /> 
        
        <rosparam file="$(find mybot_2dnav)/config/move_base_params.yaml" command="load" />
        
        <remap from="cmd_vel" to="cmd_vel"/>
        <remap from="odom" to="odom"/>
    </node>
    
    <!-- Aruco marker reachers -->
    <node if="$(eval aruco_reacher == 'simple')" pkg="mybot_2dnav" name="aruco_reacher_simple" type="aruco_reacher_simple.py" output="screen">
        
        <param name="dr_threshold" value="0.25"/>
    </node>
        
    <node if="$(eval aruco_reacher == 'action')" pkg="mybot_2dnav" name="aruco_reacher_action" type="aruco_reacher_action.py" output="screen">
        
        <param name="dr_threshold" value="0.25"/>
        <param name="dyaw_threshold" value="0.1"/>
    </node>
    
    <node if="$(eval aruco_reacher == 'search')" pkg="mybot_2dnav" name="aruco_searcher" type="aruco_searcher.py" output="screen">
        
        <param name="dr_threshold" value="0.25"/>
        <param name="dyaw_threshold" value="0.1"/>
        <param name="angle_inc" value="0.1"/>
        
    </node>
    

</launch> 
